### DISCUSSION TITLE
newbie question regarding torque

### DISCUSSION THREAD

**Discussion by RallyTronics** [DanielChappuis/reactphysics3d#259] [Q&A] [answered]
I modified the hello world code by turning off gravity and applying a local force (10, 0, 0):
```
    // Create a rigid body in the world
    Vector3 position(0, 0, 0);
    Quaternion orientation = Quaternion::identity();
    Transform transform(position, orientation);
    RigidBody* body = world->createRigidBody(transform);
    body->enableGravity(false);

    const decimal timeStep = 1.0f / 60.0f;

    // Step the simulation a few steps
    for (int i = 0; i < 100; i++) {

        body->applyLocalForceAtCenterOfMass(Vector3(10, 0, 0));
        //body->applyLocalTorque(Vector3(0, 0, 1));

        world->update(timeStep);

        // Get the updated position of the body
        const Transform& transform = body->getTransform();
        const Vector3& position = transform.getPosition();

        // Display the position of the body
        std::cout << "Body Position: (" << position.x << ", " << position.y << ", " << position.z << ")" << std::endl;
    }

```
When I run this I get what I expect
```
...
Body Position: (12.9333, 0, 0)
Body Position: (13.2028, 0, 0)
Body Position: (13.475, 0, 0)
Body Position: (13.75, 0, 0)
Body Position: (14.0278, 0, 0)

```
If I then apply a local torque 
`body->applyLocalTorque(Vector3(0, 0, 1));`

I get
```
...
Body Position: (12.245, 2.62655, 0)
Body Position: (12.472, 2.73035, 0)
Body Position: (12.6996, 2.83684, 0)
Body Position: (12.9278, 2.94604, 0)
Body Position: (13.1566, 3.05796, 0)
```
Why is there a non-zero Y component to the position?  How does applying a local torque change the direction of motion?

Thanks!

**Comment by DanielChappuis** [comment, marked as answer]
Hello. Thanks for your question.

This is expected because you are applying a local force along the **local** x direction of your body and at the same time a local torque around the z axis. Therefore, the torque that you apply will rotate your body. Next, when your local force will be applied to the **local** x axis of the body, this local axis have been rotated in the world because of the previous body rotation. Therefore, this local x axis will have a component in the direction of the world-space y axis. That's why the y component of the body position changes.

Remember that the local-space is attached to your rigid-body and will rotate with it if the body rotates.

What you probably want instead in your case is to apply a force in **world-space** along the x direction using the following method:

`RigidBody::applyWorldForceAtCenterOfMass(Vector3(10, 0, 0));` 

I hope my explanation is clear enough.
